thrifty-kotlin-codegen/src/main/kotlin/com/microsoft/thrifty/kgen/KotlinCodeGenerator.kt

```diff
@@ -501,6 +501,44 @@ class KotlinCodeGenerator(
                 .build()
     }
 
+
+    internal fun generateSealedClass(schema: Schema, struct: StructType): TypeSpec {
+        val structClassName = ClassName(struct.kotlinNamespace, struct.name)
+        val typeBuilder = TypeSpec.classBuilder(structClassName).apply {
+            addGeneratedAnnotation()
+
+            addModifiers(KModifier.SEALED)
+
+            if (struct.isDeprecated) addAnnotation(makeDeprecated())
+            if (struct.hasJavadoc) addKdoc("%L", struct.documentation)
+        }
+
+        val nameAllocator = nameAllocators[struct]
+        for (field in struct.fields) {
+            val name = nameAllocator.get(field)
+            val type = field.type.typeName
+
+            val propertySpec = PropertySpec.varBuilder("value", type.asNullable())
+                    .initializer("value")
+
+            val propConstructor = FunSpec.constructorBuilder()
+                    .addParameter("value", type.asNullable())
+
+            val dataProp = TypeSpec.classBuilder(name)
+                    .addModifiers(KModifier.DATA)
+                    .superclass(structClassName)
+                    .addProperty(propertySpec.build())
+                    .primaryConstructor(propConstructor.build())
+                    .build()
+            typeBuilder.addType(dataProp)
+        }
+
+        typeBuilder.addType(generateUnionBuilder(schema, struct))
+
+        return typeBuilder
```

Comment:
We need to add a generated `Adapter` here, and also implement the `Struct` interface.